File: BaseballApi/Contracts/BoxScoreDetail.cs
namespace BaseballApi.Contracts;

public struct BoxScoreDetail(BoxScore boxScore)
{
    public Team Team { get; set; } = boxScore.Team;

    public List<GameBatter> Batters { get; set; } = boxScore.Batters.Select(b => new GameBatter(b)).ToList();

    public List<GamePitcher> Pitchers { get; set; } = boxScore.Pitchers.Select(p => new GamePitcher(p)).ToList();

    public List<GameFielder> Fielders { get; set; } = boxScore.Fielders.Select(f => new GameFielder(f)).ToList();
}

File: baseball-app/src/app/contracts/box-score-detail.ts
import { GameBatter } from "./game-batter";
import { GameFielder } from "./game-fielder";
import { GamePitcher } from "./game-pitcher";

export interface BoxScoreDetail {
    batters: GameBatter[];
    pitchers: GamePitcher[];
    fielders: GameFielder[];
}
